COMPATIBILITY.md

```diff
@@ -0,0 +1,39 @@
+## Public API
+
+"Public API" means all public definitions in the artifacts under the
+`org.scalameta` Maven groupId whose fully-qualified name does not include
+`internal` and `contrib`, e.g. `scala.meta.Tree` or `org.langmeta.Position`:
+  * `x.y.z` must be backward compatible with `x.0.0`:
```

Comment:
I think we should make this claim stronger and guarantee compatibility against x.y1.z1 for all `y1 < y` and `z1 < z`. This will add a tax to our mima checks, maybe we can make it more digestible somehow. wdyt?

Reply:
See updates.